### PR TITLE
feat: improved workflow error handling is sdk

### PR DESCRIPTION
**Pull request by aludwiko**
References #xxxx

### CODE REVIEW

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala [hunk 1]
```diff
@@ -216,7 +220,7 @@ final class WorkflowEntityImpl(system: ActorSystem, val services: Map[String, Wo
           Future.failed(ProtocolException(command, "Receiving Workflow is not the intended recipient of command"))
 
         case InCommand(command) if command.payload.isEmpty =>
-          Future.failed(throw ProtocolException(command, "No command payload for Workflow"))
```
Comment (octonato) [inline sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala:219]: Wow, that's weird. 

Can you check the other impl? I think we copy it form ValueEntity.

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala [hunk 2]
```diff
@@ -258,8 +262,9 @@ final class WorkflowEntityImpl(system: ActorSystem, val services: Map[String, Wo
               case NonFatal(ex)       =>
                 // FIXME: not want we need.
                 // We need an exception with more context about the failed step
-                log.error(s"step execution protocol exception for $executeStep", ex)
-                throw ProtocolException(executeStep.stepName)
+                throw EntityException(
```
Comment (octonato) [inline sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala:265]: we can now remove the FIXME here above and below

**Review by octonato — APPROVED** [review]
LGTM, but we can now remove two FIXMEs

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala (hunk 1 above)
Comment (aludwiko) [reply sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala:219]: other entities are fine

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala (hunk 2 above)
Comment (aludwiko) [reply sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflowentity/WorkflowEntityImpl.scala:265]: done

**Pull request merged**